src/arr/trove/json-structs.arr

```diff
@@ -0,0 +1,102 @@
+provide *
+provide-types *
+
+import string-dict as SD
+
+data JSON:
+  | j-obj(dict :: SD.StringDict<JSON>) with:
+    native(self):
+      d = self.dict
+      ret = [SD.mutable-string-dict:]
+      for map(s from d.keys().to-list()):
+        ret.set-now(s, d.get-value(s).native())
+      end
+      ret.freeze()
+    end,
+    serialize(self):
+      d = self.dict
+      l = for map(s from d.keys().to-list()):
+        '"' + s + '": ' + d.get-value(s).serialize()
+      end
+      "{" + l.join-str(", ") + "}"
+    end
+  | j-arr(l :: List<JSON>) with:
+    native(self):
+      self.l.map(lam(x): x.native() end)
+    end,
+    serialize(self):
+      "[" + self.l.map(lam(x): x.serialize() end).join-str(", ") + "]"
+    end
+  | j-num(n :: Number) with:
+    native(self):
+      self.n
+    end,
+    serialize(self):
+      # This feels like a pretty big hack.  All I want is to convert a
+      # roughnum (or a rational) as a JavaScript number, which just
+      # means cutting off the initial "~".
```

Comment:
Yeah this suggests we should have a "num-format" for such situations.  Though with rationals, it's actually worth thinking through each situation, so...

Note that this means that going to JSON and back from a Pyret value is lossy.